src/flowchem/devices/knauer/knauer_autosampler_component.py

```diff
@@ -0,0 +1,314 @@
+        super().__init__(name, hw_device)
+        self.add_api_route("/monitor_position", self.get_monitor_position, methods=["GET"])
+        self.add_api_route("/monitor_position", self.set_monitor_position, methods=["PUT"])
+
+    def _change_connections(self, raw_position: str | int, reverse: bool = False):
+        """
+        Change connections based on the valve's raw position.
+
+        Args:
+            raw_position (int): The raw position of the valve.
+            reverse (bool): Whether to reverse the mapping.
+
+        Returns:
+            int: The mapped position.
+        """
+        position_mapping = {0: "NEEDLE", 1: "WASH", 2: "WASH_PORT2", 3: "WASTE"}
+        if reverse:
+            return str([key for key, value in position_mapping.items() if value == raw_position][0])
+        else:
+            return position_mapping[raw_position]
+
+    async def get_monitor_position(self) -> str:
+        """
+        Gets the current valve position.
+
+        Returns:
+            position (str): The current position:
+            NEEDLE (position 0).
+            WASH (position 1).
+            WASH_PORT2 (position 2).
+            WASTE (position 3).
+        """
+        position = await self.hw_device.syringe_valve_position(port=None)
+        if position:
+            logger.info(f"Syringe valve is in position: {position}")
+        return position
+
+    async def set_monitor_position(self, position: str):
+        """
+        Set the valve to a specified position.
+
+        Args:
+            position (str): The desired position:
+            NEEDLE (position 0).
+            WASH (position 1).
+            WASH_PORT2 (position 2).
+            WASTE (position 3).
+        """
+        success = await self.hw_device.syringe_valve_position(port=position)
+        if success:
+            logger.info(f"Syringe valve moved successfully to position: {position}")
+
+class AutosamplerInjectionValve(SixPortTwoPositionValve):
+    """
+    Control a Knauer Autosampler 6-Port Distribution Valve.
+
+    Attributes:
+        hw_device (KnauerValve): The hardware device for the Knauer valve.
+    """
+    hw_device: KnauerAutosampler  # for typing's sake
+    identifier = "injection_valve"
+
+    def __init__(self, name: str, hw_device: KnauerAutosampler) -> None:
+        super().__init__(name, hw_device)
+        self.add_api_route("/monitor_position", self.get_monitor_position, methods=["GET"])
+        self.add_api_route("/monitor_position", self.set_monitor_position, methods=["PUT"])
+
+    def _change_connections(self, raw_position: str | int, reverse: bool = False):
+        """
+        Change connections based on the valve's raw position.
+
+        Args:
+            raw_position (str | int): The raw position of the valve.
+            reverse (bool): Whether to reverse the mapping.
+
+        Returns:
+            str | int: The mapped position.
+        """
+        position_mapping = {0: "LOAD", 1: "INJECT"}
```

Comment:
same here, no mapping. 1 is load in the internal command structure of as and 0 is inject. no mapping needed, just follow what happens in vici valco